discord-bot/src/interactions/application_commands/matchmaking.rs

```diff
@@ -237,6 +264,153 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                
+                if let None = lobby {
+                    return Err(anyhow!(
+                        "You must run this command in a valid matchmaking thread."
+                    ));
+                }
+
+                // TODO: Refactor get resolved user from app command data to a function.
+                let resolved = data
+                    .command
+                    .resolved
+                    .ok_or_else(|| anyhow!("cannot get the resolved command user data"))?;
+                let opponent = resolved
+                    .users
+                    .values()
+                    .next()
+                    .ok_or_else(|| anyhow!("cannot get the user specified in \"report-score\" command"))?;
+                
+                let response = self.utils
+                    .http_client
+                    .channel(chan_id)
+                    .exec()
+                    .await?;
+
+                if !response.status().is_success() {
+                    return Err(anyhow!("failed to get thread"));
+                }
+
+                let channel:Channel = response
+                    .model()
+                    .await?;
+
+                let thread_members:Vec<ThreadMember> = self.utils
+                    .http_client
+                    .thread_members(channel.id)
+                    .exec()
+                    .await?
+                    .models()
+                    .await?;
+
+                let mut reporter_is_member:bool = false;
+                let mut opponent_is_member:bool = false;
+                
+                for member in thread_members {
+                    let member_user_id = member.user_id.ok_or_else(|| anyhow!("cannot get the user specified in \"report-score\" command"))?;
+                    if user.id == member_user_id {
+                        reporter_is_member = true;
+                    } else if opponent.id == member_user_id {
+                        opponent_is_member = true;
+                    } else if reporter_is_member && opponent_is_member {
+                        break;
+                    }
+                }
+
+                // TODO: Check if opponent is the bot, because that would be invalid.
+                if !reporter_is_member {
+                    return Err(anyhow!("user isn't part of the lobby"));
+                } else if !opponent_is_member {
+                    return Err(anyhow!("opponent isn't part of the lobby"));
+                }
+
+                let options:Vec<CommandDataOption>;
+                match subcommand.value.clone() {
+                    CommandOptionValue::SubCommand(x) => options = x,
+                    _ => options = Vec::<CommandDataOption>::new()
+                }
+
+                let wins_option_value = options
+                    .get(1)
+                    .ok_or_else(|| anyhow!("could not get wins option"))?
+                    .value
+                    .clone();
```

Comment:
I'm thinking we could use some better utility functions to get the command options by name. Definitely need to try to figure out if this errors.

Reply:
Yeah, I did not check if you can make it error by hand typing it but in the interactive UI it doesn't let you input invalid values or not fill an option since all are required. It probably still checks the command validity regardless but I'll check that tomorrow.